File: src/LinksNews/Services/Data/Models/Message.cs
using LinksNews.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace LinksNews.Services.Data.Models
{
    [Table("message")]
    public class Message
    {
        public long? Id { get; set; }
        public MessageGroups? MessageGroupId { get; set; }
        public long? ParentMessageId { get; set; }
        public DateTimeOffset? SentDate { get; set; }
        public string SentFromIP { get; set; }
        public long? PageId { get; set; }
        public long? AuthorAccountId { get; set; }
        public string AuthorName { get; set; }
        public string AuthorEmail { get; set; }
        public string Subject { get; set; }
        public string MessageText { get; set; }
    }
}

File: src/LinksNews/angular2App/app/core/domain/message.ts
import * as _ from 'lodash';

export class Message {
    id: number;
    messageGroupId: number;
    parentMessageId: number;
    sentDate: Date;
    sentFromIP: string;
    pageId: number;
    authorAccountId: number;
    authorName: string;
    authorEmail: string;
    subject: string;
    messageText: string;

    clone(): Message {
        let result: Message = <Message>_.assign(new Message(), this);
        return result;
    }

    static from (account: Message): Message {
        let result: Message = new Message();
        _.assign(result, account);
        return result;
    }
}
